transactor/src/test/scala/io/mediachain/transactor/JournalClusterSpec.scala

```diff
@@ -0,0 +1,172 @@
+package io.mediachain.transactor
+
+import org.specs2.specification.{AfterAll, BeforeAll}
+
+import java.util.concurrent.{BlockingQueue, LinkedBlockingQueue, TimeUnit}
+import scala.concurrent.{Future, Await}
+import scala.concurrent.duration.Duration
+import scala.concurrent.ExecutionContext.Implicits.global
+import cats.data.Xor
+
+import Types._
+
+object JournalClusterSpec extends io.mediachain.BaseSpec
+  with BeforeAll
+  with AfterAll
+{
+  val timeout = Duration(10, TimeUnit.SECONDS)
+
+  def is =
+    sequential ^
+  s2"""
+  JournalStateMachine is consistent in a cluster:
+   - the cluster can insert and update the journal $insertAndUpdateJournal
+   - all clients see the same chain mappings $checkChainViews
+   - all clients see the same current block $checkCurrentBlockViews
+   - all clients received the same journal commit events $checkCommitEvents
+   - the cluster can generate some blocks $generateBlocks
+   - all clients received the same journal block events $checkBlockEvents
+  """
+
+  def beforeAll() {
+    JournalClusterSpecContext.setup()
+  }
+  
+  def afterAll() {
+    JournalClusterSpecContext.shutdown()
+  }
+  
+  def insertAndUpdateJournal = {
+    val context = JournalClusterSpecContext.context
+    val ops = context.cluster.dummies.map(insertAndUpdate(_))
+    val res = ops.map(Await.result(_, timeout))
```

Comment:
We should probably use an XorF monad transformer for this in production (though it's probably fine here, since the timer blowing will be caught by the test harness)